MOM6/patches/mom_cap.F90.patch

```diff
@@ -1,6 +1,272 @@
++  if (chkerr(rc,__LINE__,u_FILE_u)) return
++
+   ocean_public%is_ocean_pe = .true.
++#ifdef _USE_GENERIC_TRACER
++  ! Initialise structures for extra tracer fluxes
++  call gas_exchange_init(gas_fields_atm=gas_fields_atm, gas_fields_ocn=gas_fields_ocn, gas_fluxes=gas_fluxes)
++
++  if (cesm_coupled .and. len_trim(inst_suffix)>0) then
++    call ocean_model_init(ocean_public, ocean_state, time0, time_start, gas_fields_ocn=gas_fields_ocn, &
++          input_restart_file=trim(adjustl(restartfiles)), inst_index=inst_index)
++  else
++    call ocean_model_init(ocean_public, ocean_state, time0, time_start, gas_fields_ocn=gas_fields_ocn, &
++          input_restart_file=trim(adjustl(restartfiles)))
++  endif
++
++  ! Enable data override via the data_table using the component name 'OCN'
++  call get_ocean_grid(ocean_state, ocean_grid)
++  call data_override_init(ocean_grid%Domain)
++#else
+   if (cesm_coupled .and. len_trim(inst_suffix)>0) then
+     call ocean_model_init(ocean_public, ocean_state, time0, time_start, &
+       input_restart_file=trim(adjustl(restartfiles)), inst_index=inst_index)
+   else
+     call ocean_model_init(ocean_public, ocean_state, time0, time_start, input_restart_file=trim(adjustl(restartfiles)))
+   endif
++#endif
+ 
+   ! GMM, this call is not needed in CESM. Check with EMC if it can be deleted.
+   call ocean_model_flux_init(ocean_state)
+@@ -702,6 +754,31 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
+     Ice_ocean_boundary%hcond           = 0.0
+   endif
+ 
++#ifdef _USE_GENERIC_TRACER
++  ! Allocate fields for extra tracer fluxes in Ice_ocean_boundary
++  ! Annoyingly, spawning doesn't copy param array, so add manually
++  call coupler_type_spawn(gas_fluxes, Ice_ocean_boundary%fluxes, (/isc,isc,iec,iec/), &
++        (/jsc,jsc,jec,jec/), suffix='_ice_ocn')
++  call add_gas_fluxes_param(Ice_ocean_boundary%fluxes)
++
++  ! Initialise structure for atmos fields related to extra tracer fluxes
++  ! This is set in the ESMF Internal State to be accessed elsewhere
++  ! TODO: should we deallocate atm_fields in a finalise step? Ice_ocean_boundary is handled
++  ! in a similar way and does not appear to be deallocated.
++  allocate(atm_fields)
++  ocean_internalstate%ptr%coupler_2d_bc_type_ptr => atm_fields
++  call coupler_type_spawn(gas_fields_atm, atm_fields, (/isc,isc,iec,iec/), &
++        (/jsc,jsc,jec,jec/), suffix='_atm')
++
++  ! Register diagnosics for extra tracer flux structures
++  call coupler_type_set_diags(Ice_ocean_boundary%fluxes, "ocean_flux", ocean_public%axes(1:2), time_start)
++  call coupler_type_set_diags(atm_fields, "atmos_sfc", ocean_public%axes(1:2), time_start)
++
++  ! Restore ocean fields related to extra tracer fluxes from restart files
++  call get_MOM_input(dirs=dirs)
++  call gas_fields_restore(ocean_public%fields, ocean_public%domain, dirs%restart_input_dir)
```

Comment:
Yup - this will only do anything if there is a restart file in the restart dir. Otherwise the fields will just retain their previously-initialised values.